ricecooker/utils/add_file.py

```diff
@@ -0,0 +1,204 @@
+import requests
+
+from ricecooker.classes.nodes import DocumentNode, VideoNode, TopicNode, AudioNode, HTML5AppNode
+from ricecooker.classes.files import HTMLZipFile, VideoFile, SubtitleFile, DownloadFile, AudioFile, DocumentFile, ThumbnailFile, WebVideoFile, Base64ImageFile, YouTubeSubtitleFile, YouTubeVideoFile
+from le_utils.constants import licenses
+
+from urllib.parse import urlsplit
+import hashlib
+import os
+from ricecooker.utils.transcode import transcode_video, transcode_audio
+
+class UnidentifiedFileType(Exception):
+    pass
+
+class TranscodeVideo(object):
+    "Placeholder filetype for videos that require transcoding"
+
+class TranscodeAudio(object):
+    "Placeholder filetype for audio that requires transcoding"
+
+DOWNLOAD_FOLDER = "__downloads"  # this generates completed files
+metadata = {}
+have_setup = False
+
+def setup_directory():
+    global have_setup
+    if have_setup: return
+    have_setup = True
+    try:
+        os.mkdir(DOWNLOAD_FOLDER)
+    except FileExistsError:
+        pass
+
+
+node_dict = {VideoFile: VideoNode,
+             AudioFile: AudioNode,
+             HTMLZipFile: HTML5AppNode,
+             DocumentFile: DocumentNode}
+
+# Long-Range TODOs
+# -- package up images as zip files (using build_carousel)
+
+def guess_extension(url):
```

Comment:
I think this function and `guess_type` should be moved into `ricecooker.utils.paths` as that is where our file / directory-related functions are going.

Reply:
Moved guess_extension to paths.get_extension.

For guess_type, I disagree that paths is a good place -- it only very incidently cares about filenames, mostly caring about mimetypes and file contents. I've left it as-is.